examples/scripts/linked_parameters.py

```diff
@@ -1,18 +1,40 @@
-import pybamm
+from pybamm import Parameter
 
 import pybop
 
 # The aim of this script is to show how to systematically update
 # design parameters which depend on the optimisation parameters.
 
-# Define parameter set and model
+# Define parameter set and additional parameters needed for the cost function
 parameter_set = pybop.ParameterSet.pybamm("Chen2020", formation_concentrations=True)
+parameter_set.update(
```

Comment:
Does it make sense to move these additions into BaseModel as a method called from `mass_density` and `area_density` with checks to see if these parameters are defined within the `parameter_set`?

Reply:
Adding the parameters here allows the user to provide accurate values if they have them. The aim was to remove the hidden assumptions from within the model definition. I think I will leave here for now and we can improve in future.

Reply:
Fair enough, in which case it's probably a good idea to add a check if the parameters aren't found in the `parameter_set'. At the moment, I believe a generic pybamm `key not found` error would be returned if the parameters are not available, which isn't very informative to the user.

Reply:
PyBaMM's key error is quite informative actually, e.g.

> KeyError: "'Positive electrode carbon-binder density [kg.m-3]' not found. Best matches are ['Negative electrode carbon-binder density [kg.m-3]', 'Positive electrode density [kg.m-3]', 'Positive electrode active material density [kg.m-3]']"

Let's use this

Reply:
I'm fine with this proceeding as is. Though I think the best way to solve this is to check if the user has provide a `parameter_set` with the required parameters, and if not, add them with defaults as discussed above. That's something we can update in a future PR.